sklearn/feature_extraction/hashing.py

```diff
@@ -79,6 +85,14 @@ class FeatureHasher(BaseEstimator, TransformerMixin):
     def __init__(self, n_features=(2 ** 20), input_type="dict",
                  dtype=np.float64, non_negative=False):
         self._validate_params(n_features, input_type)
+        if non_negative not in [True, False, 'total']:
+            raise ValueError("Invalid value for non_negative must be one of"
+                             " True, False, 'total'.")
+        if non_negative in ['total', True]:
+            warnings.warn("the option non_negative=True has been deprecated"
+                          " in 0.19. As of 0.21 non_negative='total' would be"
```

Comment:
Perhaps "From version 0.21, non_negative=True will be interpreted as non_negative='total'."